nexus-sdk-js/src/Resource/spec.ts

```diff
@@ -0,0 +1,234 @@
+import Resource, {
+  ResourceResponse,
+  ResourceResponseCommon,
+} from '../Resource';
+
+const mockListResponseWithStringType: ResourceResponseCommon = {
+  '@id': 'https://neuroshapes.org/commons/annotation',
+  '@type': 'https://bluebrain.github.io/nexus/vocabulary/Schema',
+  _self:
+    'https://bbp.epfl.ch/nexus/v1/schemas/anorg/testcore/nsg:commons%2Fannotation',
+  _constrainedBy:
+    'https://bluebrain.github.io/nexus/schemas/shacl-20170720.ttl',
+  _project: 'https://bbp.epfl.ch/nexus/v1/projects/anorg/testcore',
+  _createdAt: '2018-11-14T21:16:56.220Z',
+  _createdBy:
+    'https://bbp.epfl.ch/nexus/v1/realms/BBP/users/f:9d46ddd6-134e-44d6-aa74-bdf00f48dfce:sy',
+  _updatedAt: '2018-11-14T21:16:56.220Z',
+  _updatedBy:
+    'https://bbp.epfl.ch/nexus/v1/realms/BBP/users/f:9d46ddd6-134e-44d6-aa74-bdf00f48dfce:sy',
+  _rev: 1,
+  _deprecated: false,
+};
+
+const mockListResponseWithoutType: ResourceResponseCommon = {
+  '@id': 'https://incf.github.io/neuroshapes/contexts/schema.json',
+  _self:
+    'https://bbp.epfl.ch/nexus/v1/resources/anorg/testcore/https%3A%2F%2Fbluebrain.github.io%2Fnexus%2Fschemas%2Fresource.json/context:schema.json',
+  _constrainedBy: 'https://bluebrain.github.io/nexus/schemas/resource.json',
+  _project: 'https://bbp.epfl.ch/nexus/v1/projects/anorg/testcore',
+  _createdAt: '2018-11-14T21:16:54.230Z',
+  _createdBy:
+    'https://bbp.epfl.ch/nexus/v1/realms/BBP/users/f:9d46ddd6-134e-44d6-aa74-bdf00f48dfce:sy',
+  _updatedAt: '2018-11-15T08:40:52.735Z',
+  _updatedBy:
+    'https://bbp.epfl.ch/nexus/v1/realms/BBP/users/f:9d46ddd6-134e-44d6-aa74-bdf00f48dfce:sy',
+  _rev: 2,
+  _deprecated: false,
+};
+
+const mockGetByIDResponse: ResourceResponse = {
+  '@context': [
```

Comment:
Do we need the mock context to be so long?
Also, I would test the length and content of the context in the Resource instance